Modules/Core/Common/include/itkConnectedImageNeighborhoodShape.h

```diff
@@ -235,14 +221,16 @@ class ConnectedImageNeighborhoodShape
   static constexpr std::size_t
   CalculateNumberOfHypercubesOnBoundaryOfCube(const std::size_t m, const std::size_t n) ITK_NOEXCEPT
   {
+    using ::itk::Math::Experimental::UnsignedProduct;
+
     // Calculate 2^(n-m) * BinomialCoefficient(n, m)
-    return CalculateProduct(CalculatePowerOfTwo(n - m),
-                            (((2 * m) < n) ?
-                                           // Calculate either the binomial coefficient of (n, m) or (n, n - m).
-                                           // Mathematically, both should yield the same number, but the
-                                           // calculation is optimized for a smaller second argument.
-                               CalculateBinomialCoefficient(n, m)
-                                           : CalculateBinomialCoefficient(n, n - m)));
+    return UnsignedProduct(CalculatePowerOfTwo(n - m),
```

Comment:
Why not also replace `CalculatePowerOfTwo`?

Reply:
@dzenanz `CalculatePowerOfTwo` does a simple bit-shift, no need to do recursive function calls. I like that better for powers of 2. OK?